### ISSUE TITLE
It doesn't work with data providers

### ISSUE THREAD

**Issue by x-yuri** [php-vcr/phpunit-testlistener-vcr#18]
[Here][1] you check if method exists, but with data providers it's called `methodName with dataset #0`. At least if `phpunit` is called with `--filter` option.

[1]: https://github.com/php-vcr/phpunit-testlistener-vcr/blob/1.1.7/PHPUnit/Util/Log/VCR.php#L127

**Comment by x-yuri** [comment]
How to reproduce:

```sh
$ composer require php-vcr/phpunit-testlistener-vcr phpunit/phpunit
$ ./vendor/bin/phpunit --generate-configuration
$ mkdir -p tests/fixtures
```

Add

```xml
<listeners>
    <listener class="PHPUnit_Util_Log_VCR" file="vendor/php-vcr/phpunit-testlistener-vcr/PHPUnit/Util/Log/VCR.php" />
</listeners>
```

to `phpunit.xml`.

`tests/Test.php`:

```php
<?php

use PHPUnit\Framework\TestCase;
use VCR\VCR;

class MoneyTest extends TestCase
{
    static function setUpBeforeClass()
    {
        parent::setUpBeforeClass();
        VCR::configure()->enableLibraryHooks(['stream_wrapper', 'curl']);  # to not use soap
    }

    /**
     * @vcr test
     * @dataProvider dataProvider
     */
    function test1($answer)
    {
        $this->assertEquals('', substr(file_get_contents('https://google.com'), 0, 10));
    }

    function dataProvider() { return [
        [42],
    ]; }
}
```

```sh
$ ./vendor/bin/phpunit tests/Test.php --color=always
```

Change body in `tests/fixtures/test` to `''`. And add

```php
var_dump([$class, $method]);
```

to `vendor/php-vcr/phpunit-testlistener-vcr/PHPUnit/Util/Log/VCR.php:127`. Then

```sh
$ ./vendor/bin/phpunit tests/Test.php --color=always
PHPUnit 5.7.4 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.0.13 with Xdebug 2.4.1
Configuration: /home/yuri/_/1/phpunit.xml

/home/yuri/_/1/vendor/php-vcr/phpunit-testlistener-vcr/PHPUnit/Util/Log/VCR.php:127:
array(2) {
  [0] =>
  string(9) "MoneyTest"
  [1] =>
  string(22) "test1 with data set #0"
}
F                                                                   1 / 1 (100%)

Time: 540 ms, Memory: 4.00MB

There was 1 failure:

1) MoneyTest::test1 with data set #0 (42)
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-''
+'<!doctype '

/home/yuri/_/1/tests/Test.php:20

FAILURES!
Tests: 1, Assertions: 1, Failures: 1.
```

**Workaround**

```php
    function tearDown()
    {
        VCR::turnOff();
        parent::tearDown();
    }

    /**
     * @dataProvider dataProvider
     */
    function test1($answer)
    {
        VCR::turnOn();
        VCR::insertCassette('test');
        $this->assertEquals('', substr(file_get_contents('https://google.com'), 0, 10));
    }
```

**Comment by renatomefi** [comment]
@x-yuri thanks for the complete report.

This issue should be fixed on version 2.0.0. Could you please update and check again?

https://github.com/php-vcr/phpunit-testlistener-vcr/releases/tag/2.0.0

Thanks

**Comment by x-yuri** [comment]
Thanks, it's working now :)
